aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AddFIPSDualStackDecorator.kt

```diff
@@ -35,14 +40,11 @@ fun ClientCodegenContext.getBuiltIn(builtIn: Parameter): Parameter? {
  */
 fun Model.promoteBuiltInToContextParam(serviceId: ShapeId, builtInSrc: Parameter): Model {
     val model = this
-    val idx = EndpointRulesetIndex.of(model)
-    val service = model.expectShape(serviceId, ServiceShape::class.java)
-    val rules = idx.endpointRulesForService(service) ?: return model
     // load the builtIn with a matching name from the ruleset allowing for any docs updates
-    val builtIn = rules.getBuiltIn(builtInSrc) ?: return model
+    val builtIn = this.loadBuiltIn(serviceId, builtInSrc) ?: return model
 
     return ModelTransformer.create().mapShapes(model) { shape ->
-        if (shape !is ServiceShape || shape.id != service.id) {
+        if (shape !is ServiceShape || shape.id != serviceId) {
```

Comment:
Do we even need to check for the shape here? isn't checking for the ID match enough?

Reply:
checking if it's a service shape means Kotlin will infer it as a service shape in this branch